### ISSUE TITLE
Cross-Stack-Resources as a Stack

### ISSUE THREAD

**Issue by lukasniemeier-zalando** [zalando-stups/senza#151]
When setting up a new application you have to do the following steps:

1. `senza init` guides you through an interactive process which creates several *base* resources for all the following application stacks. Those may include security groups, IAM roles, etc. Additionally a senza file is created which allows you to use...
2. `senza create` to create stack versions.
3. `senza delete` to delete a stack version.

Why not have `senza init` create an actual CloudFormation stack which includes the resources created? Let's call this stack **application-id-base.yaml**. The generated **application-id.yaml** references resources from this base stack (security groups, IAM roles, ...).

1. After generating the **application-id-base.yaml** `senza init` allows the user the modify her base stack (e.g. adapting the security group configurations) - the base stack is created afterwards
2. Having the **application-id-base.yaml** allows you the easily deploy a application as-is somewhere else (see https://github.com/zalando-stups/senza/issues/98) and it allows to delete an application completely (see https://github.com/zalando-stups/senza/issues/97)

What do you think? What did I miss? :grin:

**Comment by hjacobs** [comment]
Yes, this might make sense, we just have to check how to filter (potentially) these "configuration-only" stacks (I don't want to see them every time in "senza li" output.

**Comment by lukasniemeier-zalando** [comment]
Yes filtering those would be helpful. Maybe just based on a special tag? I will look into that.

**Comment by sstirbu** [comment]
+1

**Comment by matusewicz** [comment]
+1

**Comment by lukasniemeier-zalando** [comment]
Using the changes in #183 one could define a multi-region web app template as shown in [2882d1d](https://github.com/lukasniemeier-zalando/senza/commit/2882d1d51b4f471cf59e2cfef5863b6ec17c8f41).

    # create multi region web app definition files
    senza init hello-world.yaml
    # bootstrap in eu-central-1
    senza create hello-world-base.yaml resources --region eu-central-1
    senza create hello-world.yaml 1 --region eu-central-1
    # bootstrap in eu-west-1
    senza create hello-world-base.yaml resources --region eu-west-1
    senza create hello-world.yaml 1 --region eu-west-1
    senza create hello-world.yaml 2 --region eu-west-1
    # shift traffic only in eu-west-1
    senza traffic hello-world.yaml 2 100 --region eu-west-1
    # decommission everything in eu-central-1
    senza delete hello-world.yaml 1 --region eu-central-1
    senza delete hello-world-base.yaml resources --region eu-central-1

Note: the multi-region web app template is just a crude example - might be not the best idea to add a region to a production service this way.

**Comment by lukasniemeier-zalando** [comment]
Also allows for use cases like #133 - after doing the `senza init` you are able to modify the `xxx-base.yaml`` (i.e. port configuration).
